File: translations/codenet/deepseek_commented/C++2C/starcoder/Code/s472290172.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    char str[100000];
    scanf("%s", str);
    int max = 0, cnt = 0;
    for (int i = 0; str[i]!= '\0'; i++) {
        if (str[i] == 'A' || str[i] == 'C' || str[i] == 'G' || str[i] == 'T') {
            cnt++;
        } else {
            cnt = 0;
        }
        max = max < cnt? cnt : max;
    }
    printf("%d\n", max);
}

File: translations/codenet/uncommented/C++2J/gpt-4o/Code/s472290172.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String str = scanner.next();
        
        int max = 0, cnt = 0;
        for (char sc : str.toCharArray()) {
            if (sc == 'A' || sc == 'C' || sc == 'G' || sc == 'T') {
                cnt++;
            } else {
                cnt = 0;
            }
            max = Math.max(max, cnt);
        }
        
        System.out.println(max);
        scanner.close();
    }
}

// <END-OF-CODE>
